### DISCUSSION TITLE
Plotting spatially embedded tree graphs

### DISCUSSION THREAD

**Discussion by NikDrummond** [marcomusy/vedo#895] [General]
I'm just wondering how i would go about plotting a tree graph where the points are embedded in 3D space, so have coordinates already. I've seen the DirectedGraph stuff, but I don't want to use a layout. I tried generating a mesh with vertices and edges, and passing that to Mesh, but this just crashes - I assume because the edges are not faces, and a tree has no cycles. 

If anyone has any ideas thank you!

**Comment by marcomusy** [comment]
Hi, if you have a set of vertices and a list of indices then you can use `Lines`, eg:
```python
from vedo import *

vertices = np.random.randn(10, 3)
edges = [[0,2], [1,2], [2,3], [3,4], [4,5], [5,6], [6,7]]

lns = Lines(vertices[edges], c="red5", lw=3)
pts = Points(vertices, r=10, c="green5")
labels = pts.labels2d("id", font="Quikhand", scale=1.2)
show(pts, labels, lns, axes=1)
```
![Screenshot from 2023-07-13 23-00-02](https://github.com/marcomusy/vedo/assets/32848391/ebc37696-e49f-42ee-8b19-35145970dbaf)
